### ISSUE TITLE
HDF5 not properly linked on Windows + Bash + Ninja

### ISSUE THREAD

**Issue by tdegeus** [BlueBrain/HighFive#422]
I am trying to link HDF5 using CMake as follows:

```cmake
set(test_name "unit-tests")
set(HIGHFIVE_USE_BOOST 0)
set(HIGHFIVE_USE_XTENSOR 1)
find_package(HighFive REQUIRED)
target_link_libraries(${test_name} PRIVATE HighFive)
```

Then, on my GitHub action:

```yaml
    defaults:
      run:
        shell: bash -l {0}

    name: "${{ matrix.runs-on }} • x64 ${{ matrix.args }}"
    runs-on: ${{ matrix.runs-on }}

    steps:

    ...

    - name: Configure using CMake
      run: cmake -Bbuild -G Ninja

    - name: Build C++
      working-directory: build
      run: cmake --build .
```

With this I'm getting the error that

```
Linking CXX executable test\unit-tests.exe
FAILED: test/unit-tests.exe 
cmd.exe /C "cd . && C:\Miniconda\envs\myenv\Library\bin\clang++.exe -fuse-ld=lld-link -nostartfiles -nostdlib -O3 -DNDEBUG -D_DLL -D_MT -Xclang --dependent-lib=msvcrt  test/CMakeFiles/unit-tests.dir/main.cpp.obj test/CMakeFiles/unit-tests.dir/Redraw.cpp.obj test/CMakeFiles/unit-tests.dir/Static.cpp.obj -o test\unit-tests.exe -Xlinker /implib:test\unit-tests.lib -Xlinker /pdb:test\unit-tests.pdb -Xlinker /version:0.0   C:/Miniconda/envs/myenv/Library/lib/libhdf5.lib  -lkernel32 -luser32 -lgdi32 -lwinspool -lshell32 -lole32 -loleaut32 -luuid -lcomdlg32 -ladvapi32 -loldnames && cd ."
lld-link: error: undefined symbol: __declspec(dllimport) H5Pclose
```
(and many more of these linker errors).

For full reference : https://github.com/tdegeus/QPot/pull/19

**Comment by PhilipDeegan** [comment]
You might be able to get around this by adding a define for 

-DH5_BUILT_AS_DYNAMIC_LIB=1

https://github.com/HDFGroup/hdf5/blob/2ea165efd095d8876639814dc615dd01a2ec34e6/src/H5api_adpt.h#L23

**Comment by tdegeus** [comment]
Thanks for the suggestion @PhilipDeegan  , but unfortunately that did not fix... But it does suggest that I might be understanding things incorrectly. I thought that HDF5 might not be linked correctly, but your suggestion implies that it might be linked correctly, but that it might be missing some symbols because of the way HDF5 was coded?

**Comment by tdegeus** [comment]
What is a bit strange is that I don't find something like `-lhdf5` as argument, wheres the `HighFive` target clear does something:

```

**Comment by PhilipDeegan** [comment]
you have 

    C:/Miniconda/envs/myenv/Library/lib/libhdf5.lib

if you have no dll, you might only have a static library version of hdf5

**Comment by tdegeus** [comment]
The problems curiously disappeared from me. This might be due to a HDF5 and/or CMake update

**Comment by tdegeus** [comment]
And curiously reappeared here: https://github.com/tdegeus/XDMFWrite_HighFive/pull/11

**Comment by 1uc** [comment]
We've rewritten the HighFive CMake related code for `v3` and are close to of have merge the changes. I'm closing this issue as outdated. Please open a new Issue if the problem persists after #897 got merged.
